docs/installation/server-config/security.rst

```diff
@@ -1,4 +1,242 @@
+the principal is the user the KSQL server has authenticated as, with the Apache Kafka cluster, or an appropriate group
+that includes the authenticated KSQL user.
+
+.. tip:: For more information about ACLs see :ref:`kafka_authorization` and for more information about interactive and
+         non-interactive queries, see :ref:`restrict-ksql-interactive`.
+
+-------------------------
+Interactive KSQL clusters
+-------------------------
+
+:ref:`Interactive KSQL clusters <restrict-ksql-interactive>` require these ACLs:
+
+- Permission for the ``DESCRIBE_CONFIGS`` operation on the ``CLUSTER`` resource type.
+- Permission for the ``CREATE`` operation on the ``CLUSTER`` resource type.
+- Permissions for ``DESCRIBE``, ``READ``, ``WRITE`` and ``DELETE`` operations on all ``TOPIC`` resource types.
+- Permissions for ``DESCRIBE`` and ``READ`` operations  on all ``GROUP`` resource types.
+
+It is still possible to restrict the authenticated KSQL user from accessing specific resources using ``DENY`` ACLs. For
+example, you can add a ``DENY`` ACL to stop KSQL queries from accessing a topic that contains sensitive data.
+
+----------------------------------------
+Non-Interactive (headless) KSQL clusters
+----------------------------------------
+
+:ref:`Non-interactive KSQL clusters <restrict-ksql-interactive>` can be run with much more restrictive ACLs, though it
+currently requires a bit of effort to determine what ACLs are required. This will be improved in future KSQL releases.
+
+Standard ACLs
+    The authenticated KSQL user always requires:
+
+    - ``DESCRIBE_CONFIGS`` permission on the ``CLUSTER`` resource type.
+    - ``DESCRIBE`` permission on the ``__consumer_offsets`` topic.
+
+Input topics
+    The authenticated KSQL user requires ``DESCRIBE`` and ``READ`` permissions for each input topic. The topic should already exist
+    when KSQL is started.
+
+Output topics
+    The authenticated KSQL user requires ``DESCRIBE`` and ``WRITE`` permissions ror each output topic.
+
+    By default, KSQL will attempt to create any output topics that do not exist. To allow this the authenticated KSQL user requires
+    ``CREATE`` permissions on the ``CLUSTER`` resource type. Alternatively, topics can be created manual before running KSQL. To determine
+    the list of output topics and their required configuration, for example, partition count, replication factor,
+    retention policy, etc, you can run KSQL on an open cluster first.
+
+    If you want KSQL to create output topics, then the authenticated KSQL user should be granted ``CREATE`` permission on the ``CLUSTER`` resource type. KSQL
+    will create output topics whenever you are doing a persistent CTAS/CTAS query.
+
+Change-log and repartition topics
+    Internally KSQL uses repartition topics and changelog topics for selected operations. The authenticated KSQL user minimally requires a
```

Comment:
Think this reads better as:

Internally KSQL uses repartition topics and changelog topics for selected operations. KSQL requires repartition topics when using either PARTITION BY, or using GROUP BY on non-key values, and requires changelog topics for any CREATE TABLE x AS statements.

The authenticated KSQL user minimally requires a DESCRIBE, READ, and WRITE permissions for each changelog and repartition the TOPIC resource type.

In addition, KSQL will attempt to create any repartition or changelog topics that do not exist. To allow this, the authenticated KSQL user requires CREATE permissions on the CLUSTER resource type. Alternatively, topics can be created manually before running KSQL. To determine the list of topics and their required configuration, for example, partition count, replication factor, retention policy, etc, you can run KSQL on an open cluster first.

All changelog and repartition topics are prefixed with _confluent-ksql-<value of ksql.service.id property>_query_<query id>_ where ksql.service.id defaults to ksql_.